packages/components/pie-toast/src/index.ts

```diff
@@ -119,6 +218,14 @@ export class PieToast extends RtlMixin(LitElement) implements ToastProps {
     protected async updated (_changedProperties: PropertyValues<this>) {
         if (_changedProperties.has('isOpen') && this.isOpen) {
             dispatchCustomEvent(this, ON_TOAST_OPEN_EVENT, { targetNotification: this });
+
+            if (this.duration !== null) {
+                this.createAutoDismissEventListeners();
+            }
```

Comment:
The duration check could probably be moved inside the method?

Reply:
It depends, IMHO leave the check out of it will help to understand that event listeners will be created only when the duration `is not null` at first glance.